src/cloud-api-adaptor/test/provisioner/provision.go

```diff
@@ -287,46 +287,77 @@ func (lio *KbsInstallOverlay) Edit(ctx context.Context, cfg *envconf.Config, pro
 	return nil
 }
 
-func (p *KeyBrokerService) GetKbsPodIP(ctx context.Context, cfg *envconf.Config) (string, error) {
+func getNodeIPForSvc(service corev1.Service, cfg *envconf.Config) (string, error) {
 	client, err := cfg.NewClient()
 	if err != nil {
 		return "", err
 	}
+	podList := &corev1.PodList{}
+	if err := client.Resources(service.Namespace).List(context.TODO(), podList); err != nil {
+		return "", err
+	}
 
-	namespace := "coco-tenant"
-	deploymentName := "kbs"
+	nodeList := &corev1.NodeList{}
+	if err := client.Resources("").List(context.TODO(), nodeList); err != nil {
+		return "", err
+	}
 
-	err = AllPodsRunning(ctx, cfg, namespace)
+	for _, pod := range podList.Items {
+		if pod.Spec.NodeName != "" {
```

Comment:
or shall we find the KBS pod first here but not use the 1st pod?

Reply:
changed to filter pod first on deployment name.